### PR TITLE
ocamlPackages.stringext: 1.4.0 -> 1.6.0

### PR DESCRIPTION
**Pull request by ulrikstrid**
###### Motivation for this change

Upstreaming packages from overlays

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/stringext/default.nix [hunk 1]
```diff
@@ -1,28 +1,21 @@
-{ stdenv, lib, fetchzip, ocaml, findlib, ocamlbuild, ounit, qcheck
+{ stdenv, lib, fetchurl, ocaml, buildDunePackage, ounit, qtest
 # Optionally enable tests; test script use OCaml-4.01+ features
 , doCheck ? lib.versionAtLeast (lib.getVersion ocaml) "4.01"
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/stringext/default.nix:3]:
```suggestion
, doCheck ? lib.versionAtLeast ocaml.version "4.04"
```

File: pkgs/development/ocaml-modules/stringext/default.nix [hunk 2]
```diff
@@ -1,28 +1,21 @@
-{ stdenv, lib, fetchzip, ocaml, findlib, ocamlbuild, ounit, qcheck
+{ stdenv, lib, fetchurl, ocaml, buildDunePackage, ounit, qtest
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/stringext/default.nix:1]:
```suggestion
{ lib, fetchurl, ocaml, buildDunePackage, ounit, qtest
```

File: pkgs/development/ocaml-modules/stringext/default.nix [hunk 3]
```diff
@@ -1,28 +1,21 @@
...
 }:
 
-let version = "1.4.3"; in
+let version = "1.6.0"; in
 
-stdenv.mkDerivation {
-  name = "ocaml${ocaml.version}-stringext-${version}";
+buildDunePackage {
+  pname = "stringext";
+  version = version;
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/stringext/default.nix:11]:
```suggestion
  useDune2 = true;
```

**Review by vbgl — COMMENTED** [review]
Also please remove the `meta.platforms` attribute: it is set by `buildDunePackage`.

**Comment by ulrikstrid** [comment]
Thank you for the review @vbgl !

**Pull request merged**